metricflow/time/time_granularity_solver.py

```diff
@@ -253,6 +268,80 @@ def adjust_time_range_to_granularity(
 
         return TimeRangeConstraint(start_time=constraint_start, end_time=constraint_end)
 
+    def _get_primary_time_dimension(self) -> TimeDimensionSpec:  # noqa: D
+        for data_source in self._semantic_model.user_configured_model.data_sources:
+            for dimension in data_source.dimensions:
+                if dimension.type_params:
+                    if dimension.type_params.time_granularity:
+                        if dimension.type_params.is_primary:
+                            return TimeDimensionSpec.parse(dimension.name.element_name)
```

Comment:
Let's not duplicate this dimension walk here. You can do the following, or you can just delete this method and inline the conversion to TimeDimensionSpec down below in the one place where you use this.

```suggestion
        primary_time_dimension_reference = self._semantic_model.data_source_semantics.primary_time_dimension_reference:
        return TimeDimensionSpec.parse(primary_time_dimension_reference.element_name)
```